### ISSUE TITLE
Success probability for small blocksize

### ISSUE THREAD

**Issue by lducas** [fplll/strategizer#7]
@yuyang-Tsinghua Notified me that when one want to run BKZ with small blocksize and high success probability, the current strategy sucks, because restarting LLL is just so slow compared to the enumeration itself. The target success probability should be set much higher.

I.e. even if used with the default .5 later on, one can run the pruner with target proba .95 for example. 
It may even be done for large blocksize without affecting the efficiency, shouldn't it ?

**Comment by lducas** [comment]
Can't assign myself for some reason. Also, I could use some help on where the success proba is set.

**Comment by malb** [comment]
Apparently, I can assign you :) Success probability is set here: 

https://github.com/fplll/strategizer/blob/master/strategize.py#L126

Happy to help.

**Comment by yuyang-crypto** [comment]
I appreciate your help both. I am still a little confused.  If I need to obtain a nice basis (with min_success_probability 0.95 which I want), should I set the min_success_probability 0.95 in your new uploaded file "strategize.py"?

**Comment by lducas** [comment]
Hi Yang,

strategize.py will just optimize pruing parameter for a given use. It will give a base-success-probability, but you can still call fplll/fpylll with different success probability, in which case fplll will resort to retrials. They seems way too expensive in small blocksize, so this is why I think strategize.py should produce strategies with a very high base-success-proba. I hope I can do that in the next few days.

**Comment by yuyang-crypto** [comment]
Hi Leo,

I have obtained an optimized pruning parameter with "base-success-probability" 0.95 (set in the file strategize.py before generating, right?). When blocksize < 49, the target probability of strategy is always greater than 0.95, so the result would be same when I set min_success_probability either 0.5 or 0.95 in fplll/fpylll. But for larger blocksizes, the smallest target probability would decrease gradually. Thus, to obtain a high-quality basis, I still need set  a high "min_success_probability" in fpylll for those large blocksizes. In this situation, the main cost would be enumeration, hence pruning would be very efficient. Is it right?
